api/context.md

```diff
@@ -162,16 +162,18 @@ If you want to create the middleware which provides a custom method,
 write like the following:
 
 ```ts
-const app = new Hono()
-
-const echoMiddleware: MiddlewareHandler<{
+type Env = {
   Variables: {
     echo: (str: string) => string
   }
-}> = async (c, next) => {
+}
+
+const app = new Hono<Env>()
```

Comment:
It does not need passing `Env` as a generics:

```ts
const app = new Hono()
```

Reply:
As pointed out in the referenced issue, this PR is to solve the problem of type completion when using middleware with use.
As shown in the following image, this type problem can be solved by passing Env to Hono's generics.
<img width="623" alt="Screenshot 2024-04-29 at 19 00 44" src="https://github.com/honojs/website/assets/80559385/f926f89f-138f-4fec-bd7a-019373b10363">

<img width="640" alt="Screenshot 2024-04-29 at 19 01 01" src="https://github.com/honojs/website/assets/80559385/b9ad2ad8-10a1-45fd-a567-75f7a8c0efee">

Reply:
Should I write this as a separate section?

Reply:
I made `createMiddleware()` to make the user not write `Hono<Env>` if added just before the handler. So I don't want to write `Hono<Env>` if it is not necessary.

Then, you can write, `If you want to add your custom middleware separate from handlers, you should pass the `Env` to Hono as generics`, as another example.

Reply:
Okay, I’ll separate the codes. Thank you

Reply:
Gooood! Thanks!